charmstore.go

```diff
@@ -49,6 +50,12 @@ type NewCharmStoreParams struct {
 	// the user visits a web page to authenticate themselves.
 	// If nil, a default function that returns an error will be used.
 	VisitWebPage func(url *url.URL) error
+
+	// User and Password hold the authentication credentials
+	// for the client. If User is empty, no credentials will be
+	// sent.
```

Comment:
I think you need to separate this comment to be above their respective member variables.